### ISSUE TITLE
Linkerd-tcp panics during redis load-test

### ISSUE THREAD

**Issue by smparekh** [linkerd/linkerd-tcp#56]
Error message:
```
thread 'main' panicked at 'destination and source slices have different lengths', /buildslave/rust-buildbot/slave/stable-dist-rustc-linux/build/src/libcore/slice.rs:533
```

Seeing this during a very high load-test of redis (batch inserting 5000 keys per second). The linkerd-tcp container does restart after throwing this message and I am seeing some keys being dropped in redis. I will try to recreate this with `RUST_BACKTRACE=1`.

**Comment by stevej** [comment]
Is this from master or from a release? If it's from a release, then I believe this is fixed in master with #48

**Comment by smparekh** [comment]
It is from a release (0.3).

Sorry for duplicate posting, I glanced through to the issues but missed the one you linked. I will verify this in the next release.
